server/internal/usecase/interactor/request.go

```diff
@@ -0,0 +1,213 @@
+package interactor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase"
+	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
+	"github.com/reearth/reearth-cms/server/internal/usecase/interfaces"
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/request"
+	"github.com/reearth/reearth-cms/server/pkg/thread"
+	"github.com/reearth/reearth-cms/server/pkg/version"
+	"github.com/reearth/reearthx/usecasex"
+)
+
+type Request struct {
+	repos    *repo.Container
+	gateways *gateway.Container
+}
+
+func NewRequest(r *repo.Container, g *gateway.Container) *Request {
+	return &Request{
+		repos:    r,
+		gateways: g,
+	}
+}
+
+func (r Request) FindByID(ctx context.Context, id id.RequestID, operator *usecase.Operator) (*request.Request, error) {
+	return r.repos.Request.FindByID(ctx, id)
+}
+
+func (r Request) FindByIDs(ctx context.Context, list id.RequestIDList, operator *usecase.Operator) ([]*request.Request, error) {
+	return r.repos.Request.FindByIDs(ctx, list)
+}
+
+func (r Request) FindByProject(ctx context.Context, pid id.ProjectID, filter interfaces.RequestFilter, pagination *usecasex.Pagination, operator *usecase.Operator) ([]*request.Request, *usecasex.PageInfo, error) {
+	return r.repos.Request.FindByProject(ctx, pid, repo.RequestFilter{
+		State:   filter.State,
+		Keyword: filter.Keyword,
+	}, pagination)
+}
+
+func (r Request) Create(ctx context.Context, param interfaces.CreateRequestParam, operator *usecase.Operator) (*request.Request, error) {
+	if operator.User == nil {
+		return nil, interfaces.ErrInvalidOperator
+	}
+
+	return Run1(ctx, operator, r.repos, Usecase().Transaction(), func() (*request.Request, error) {
+		p, err := r.repos.Project.FindByID(ctx, param.ProjectID)
+		if err != nil {
+			return nil, err
+		}
+		ws, err := r.repos.Workspace.FindByID(ctx, p.Workspace())
+		if err != nil {
+			return nil, err
+		}
+
+		th, err := thread.New().NewID().Workspace(ws.ID()).Build()
+
+		if err != nil {
+			return nil, err
+		}
+		if err := r.repos.Thread.Save(ctx, th); err != nil {
+			return nil, err
+		}
+
+		builder := request.New().
+			NewID().
+			Workspace(ws.ID()).
+			Project(param.ProjectID).
+			CreatedBy(*operator.User).
+			Thread(th.ID()).
+			Items(param.Items).
```

Comment:
- All request.Items in params.Item must have the pointer set to public.
- Get items from `repos.Item.FindByIDs`, and then if current latest version is the same as public ref. a new request cannot be created. Return error.